sdk/state/open.go

```diff
@@ -47,18 +56,23 @@ func (c *Channel) OpenTxs() (txClose, txDecl, formation *txnbuild.Transaction, e
 		InitiatorEscrow: c.initiatorEscrowAccount().Address,
 		ResponderEscrow: c.responderEscrowAccount().Address,
 		StartSequence:   c.startingSequence,
-		// TODO - change to non native
-		Asset: NativeAsset{},
+		Asset:           p.Asset,
+		AssetLimit:      p.AssetLimit,
 	})
 	return
 }
 
 // ProposeOpen proposes the open of the channel, it is called by the participant
 // initiating the channel.
-func (c *Channel) ProposeOpen() (Open, error) {
+func (c *Channel) ProposeOpen(p OpenParams) (Open, error) {
+	if !p.Asset.IsNative() {
```

Comment:
need to validate that asset limit is a string that can be parsed as an int. Note it's a type string because `txnbuild.ChangeTrust` operation takes `Limit` as a string input

Reply:
Generally I prefer to keep things strongly typed until the point of translation where possible. Puts more work on the compiler so less context to keep in the programmer's mind while coding.

That would also eliminate us needing to parse as a String here.
But fine the way it is as a string for now so no need to change.